ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -123,10 +130,18 @@ public boolean valueContainsAny(final Set<ColumnName> names) {
    *
    * @param windowed indicates that the source is windowed; meaning {@code WINDOWSTART} and {@code
    * WINDOWEND} columns will added to the value schema to represent the window bounds.
+   *
+   * @param pseudoColumnVersion determines the set of pseudocolumns to be added to the schema
+   *
    * @return the new schema.
    */
+  public LogicalSchema withPseudoAndKeyColsInValue(
+      final boolean windowed, final int pseudoColumnVersion) {
+    return rebuildWithPseudoAndKeyColsInValue(windowed, pseudoColumnVersion);
+  }
+
   public LogicalSchema withPseudoAndKeyColsInValue(final boolean windowed) {
-    return rebuild(true, windowed);
+    return rebuildWithPseudoAndKeyColsInValue(windowed, CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
```

Comment:
nit: let's have this method call the overload, rather than having both call `rebuildWithPseudoAndKeyColsInValue()`. IMO code is easier to read this way since it makes the relationship between the overloads more clear.